apps/server/src/modules/common-cartridge/service/common-cartridge-export.service.ts

```diff
@@ -201,40 +202,63 @@ export class CommonCartridgeExportService {
 			title: card.title ?? '',
 			identifier: createIdentifier(card.id),
 		});
+		const fileMetadataBufferArray: { id: string; name: string; fileBuffer: Buffer; fileDto: FileDto }[] = [];
 
-		await Promise.all(card.elements.map((element) => this.addCardElementToOrganization(element, cardOrganization)));
+		await Promise.all(card.elements.map((element) => this.downloadAndStoreFiles(element, fileMetadataBufferArray)));
+		await Promise.all(
+			card.elements.map((element) =>
+				this.addCardElementToOrganization(element, cardOrganization, fileMetadataBufferArray)
+			)
+		);
 	}
 
-	private async addCardElementToOrganization(
+	private async downloadAndStoreFiles(
 		element: CardResponseElementsInnerDto,
-		cardOrganization: CommonCartridgeOrganizationNode
+		fileMetadataBufferArray: { id: string; name: string; fileBuffer: Buffer; fileDto: FileDto }[]
 	): Promise<void> {

```

Comment:
use the created type to return the data instead of a shared array across async functions.

```suggestion
	private async downloadAndStoreFiles(
		element: CardResponseElementsInnerDto
	): Promise<CREATEDTYPE> {
```

Reply:
All the data of the array is created here. Not using the shared array and using a return value instead makes this easier to understand.

Reply:
Modified